app/adapters/alma_adapter/availability_status.rb

```diff
@@ -12,61 +12,71 @@ def initialize(bib:)
+      end
+      availability
+    end
+
+    def holding_status(holding:, sequence:)
+      # Ignore electronic and digital records
+      return nil if holding["inventory_type"] != "physical"
+
       status_label = Status.new(bib: bib, holding: holding, holding_item_data: nil).to_s
-      status = if holding["holding_id"]
-                 {
-                   on_reserve: "N",
-                   location: holding["library_code"] + "$" + holding["location_code"],
-                   label: holding["location"],
-                   status_label: status_label,
-                   cdl: false,
-                   holding_type: "physical",
-                   id: holding["holding_id"]
-                 }
-               elsif holding["portfolio_pid"]
-                 # This kind of data is new with Alma.
-                 {
-                   on_reserve: "N",
-                   location: "N/A",
-                   label: "N/A",
-                   status_label: status_label,
-                   cdl: false,
-                   holding_type: "portfolio",
-                   id: holding["portfolio_pid"]
-                 }
-               else
-                 # TODO: can there be more than one like this per bib?
-                 # If so we'll need to use unique IDs here.
-                 {
-                   on_reserve: "N",
-                   location: holding["location_code"],
-                   label: holding["location"],
-                   status_label: status_label,
-                   cdl: false,
-                   holding_type: "other",
-                   id: "other"
-                 }
-               end
-
-      # Check if the item is available via CDL.
-      # Notice that we only do this when necessary because it requires an extra (slow-ish) API call.
-      check_cdl = status[:holding_type] == "physical" && status[:status_label] == "Unavailable"
-      status[:cdl] = cdl_holding?(holding["holding_id"]) if check_cdl
+      status = {
+        on_reserve: "N",
+        location: holding["library_code"] + "$" + holding["location_code"],
+        label: holding["location"],
+        status_label: status_label,
+        copy_number: nil,
+        cdl: false,
+        temp_location: holding["holding_id"].nil?,
+        id: holding["holding_id"]
+      }
 
+      # Some physical resources can have a nil ID when they are in a temporary location
+      # because Alma does not tells us the holding_id that they belong to. For those
+      # records we create a fake ID so that we can handle multiple holdings with this
+      # condition.
+      status[:id] = "fake_id_#{sequence}" if holding["holding_id"].nil?
+
+      # Notice that we only check if a holding is available via CDL when necessary
+      # because it requires an extra (slow-ish) API call.
+      status[:cdl] = cdl_holding?(holding["holding_id"]) if status[:status_label] == "Unavailable"
+
+      status
+    end
+
+    def holding_status_from_item(alma_item)
```

Comment:
This method is new. 

This method builds a response that is identical to what the method `holding_status` (just above this one) returns.

This new method builds the response from the data coming from Alma's `/bib/{id}/holdings/all/items` API as opposed to the information in the AVA fields in the Alma's `/bib/{id}` endpoint.